File: pdu/python/std_msgs/pdu_cdr_conv_Int32.py
from .pdu_pytype_Int32 import Int32
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_Int32(writer: CdrWriter, src: Int32):
    writer.write_int32(src.data)


def cdr_body_to_py_Int32(reader: CdrReader, dst: Int32):
    dst.data = reader.read_int32()
    return dst


def py_to_cdr_Int32(src: Int32) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Int32(writer, src)
    return writer.bytes()


def cdr_to_py_Int32(cdr_payload) -> Int32:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Int32()
    return cdr_body_to_py_Int32(reader, dst)

File: pdu/javascript/std_msgs/pdu_cdr_conv_Int32.js
import { Int32 } from './pdu_jstype_Int32.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduInt32Converter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Int32 } src
     */
    static to_cdr_body(writer, src) {
        writer.write_int32(src.data);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Int32 } dst
     * @returns { Int32 }
     */
    static cdr_body_to_js(reader, dst) {
        dst.data = reader.read_int32();
        return dst;
    }

    /**
     * @param { Int32 } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Int32 }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Int32());
    }
}
